triggers.md

```diff
@@ -0,0 +1,235 @@
+## Motivation
+
+Users want to programatically trigger pipelines.
+
+The default in Pachyderm is that pipelines eagerly process data that's
+committed to their inputs. This should remain the default but it breaks down in
+some use cases. For those use cases we have the pattern of deferred processing,
+this allows you to fully control when pipelines run by moving branches. This is
+also something that should remain an option, but it also breaks down in some
+use case. The motivation for triggers is to add something between these two
+options that doesn't require a separate process (or a human) to move the branch
+heads but also doesn't move them blindly on every commit.
+
+## Requirements
+
+For the purposes of this rfc we'll limit the trigger conditions to the
+following metadata aspects:
+
+- cron: trigger a pipeline if a cron spec is satisfied and new data has been committed.
+- size: trigger a pipeline if an input has grown by a given amount.
+- commits: trigger a pipeline if input has received a given number of commits.
+
+The only trigger that might have made this list but didn't is number of files.
+This wouldn't be terribly hard to add, the only downside is that it requires
+walking the files which would make it less performant than the other triggers.
+
+It is not possible for a trigger to set more than one of these aspects.
```

Comment:
Can you explain why not? What if a user wants to trigger every 10MB or every 5 mins with 10MB hasn't come in by then? Sam's wants her deferred processes to do exactly this. As stated, that may be in the "do it yourself with branches" type of situation, but I'd like to understand what about this design specifically prevents someone from maybe wanting to do nested triggers like this?

Reply:
It's not hard to do this implementation wise, but it takes us down to the path to building a DSL. The obvious other ask if you can do 10M _or_ every 5 mins is can you do 10M _and_ every 5 mins? Both seem like they'd be pretty useful to me. I was trying to simplify as much as possible in the initial proposal, so I didn't include either one, but if you feel like they're necessary I'm fine adding them.